packages/app/src/cli/models/app/app.ts

```diff
@@ -98,56 +99,63 @@ const WebhooksSchemaWithDeclarative = WebhooksSchema.merge(DeclarativeWebhooksSc
   }
 })
 
-export const AppSchema = zod
-  .object({
-    name: zod.string().max(30),
-    client_id: zod.string(),
-    application_url: validateUrl(zod.string()),
-    embedded: zod.boolean(),
-    access_scopes: zod
-      .object({
-        scopes: zod.string().optional(),
-        use_legacy_install_flow: zod.boolean().optional(),
-      })
-      .optional(),
-    auth: zod
-      .object({
-        redirect_urls: zod.array(validateUrl(zod.string())),
-      })
-      .optional(),
-    access: zod
-      .object({
-        direct_api_offline_access: zod.boolean().optional(),
-      })
-      .optional(),
-    webhooks: WebhooksSchemaWithDeclarative,
-    app_proxy: zod
-      .object({
-        url: validateUrl(zod.string()),
-        subpath: zod.string(),
-        prefix: zod.string(),
-      })
-      .optional(),
-    pos: zod
-      .object({
-        embedded: zod.boolean(),
-      })
-      .optional(),
-    app_preferences: zod
-      .object({
-        url: validateUrl(zod.string().max(255)),
-      })
-      .optional(),
-    build: zod
-      .object({
-        automatically_update_urls_on_dev: zod.boolean().optional(),
-        dev_store_url: zod.string().optional(),
-      })
-      .optional(),
-    extension_directories: zod.array(zod.string()).optional(),
-    web_directories: zod.array(zod.string()).optional(),
-  })
-  .strict()
+export const NonVersionedAppTopSchema = zod.object({
+  name: zod.string().max(30),
+  client_id: zod.string(),
+})
```

Comment:
Just to avoid modifyig  the position of the fields inside the `toml` file. 

The final schema is created in this sentence 
```ts
export const AppSchema = NonVersionedAppTopSchema.merge(VersionedAppSchema).merge(NonVersionedAppBottomSchema).strict()
```
 That way when the config is written to the file it will be done in the same order.

Once the `name`, `access_scopes` and `auth` are versioned in a future PR we can simplify this to have just a `NonVersionedAppSchema`